### PR TITLE
MC-36779: Product still present in the Wish List after added to order

### LINKED ISSUE #30260: Product still present in the Wish List after added to order
Preconditions:
 1. Magento2ee 2.4.0-p1 with b2b is installed and should be actual for CE
 2. Create simple product
 3. Create Customer
 Steps:

1. Go to admin -> sales -> orders -> Created New Order -> Select Customer
 2. Add simple product to the Order
 3 Select Action => Move to Wish List
 4. click Update Items and Quantities button
 5. select checkbox 'Add To Order' => yes
 6. click Update Changes click

(/) Expected Result:
 Product moved to Order
 Wish List is empty

(x) Actual Result:
 Product moved to Order
 Wish List is not empty

### PR DESCRIPTION
**Pull request by SmVladyslav**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#30260 (MC-36779)

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by SmVladyslav** [comment]
@magento run all tests

File: app/code/Magento/Sales/Model/AdminOrder/Create.php [hunk 1]
```diff
@@ -970,7 +970,7 @@ public function applySidebarData($data)
                 $item = $this->getCustomerCart()->getItemById($itemId);
                 if ($item) {
                     $this->moveQuoteItem($item, 'order', $qty);
-                    $this->removeItem($itemId, 'cart');
+                    $data['remove'][$itemId] = 'cart';
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/Model/AdminOrder/Create.php]: What if `$data['remove']` does not exist?

File: app/code/Magento/Sales/Model/AdminOrder/Create.php [hunk 2]
```diff
@@ -984,6 +984,7 @@ public function applySidebarData($data)
                 );
                 if ($item->getId()) {
                     $this->addProduct($item->getProduct(), $item->getBuyRequest()->toArray());
+                    $data['remove'][$itemId] = 'wishlist';
```
Comment (lbajsarowicz) [inline app/code/Magento/Sales/Model/AdminOrder/Create.php]: What if `$data['remove']` does not exist?

**Review by lbajsarowicz — COMMENTED** [review]
🟡  @SmVladyslav I see that you reference the array key, but as far as I understand - there's no guarantee this key will exist. Isn't that necessary to check whether the key exist? (eg. `isset()`)

**Comment by lbajsarowicz** [comment]
@magento run Functional Tests CE, Functional Tests B2B, Unit Tests, WebAPI Tests

**Comment by SmVladyslav** [comment]
@magento run all tests

File: app/code/Magento/Sales/Model/AdminOrder/Create.php (hunk 1 above)
Comment (SmVladyslav) [reply app/code/Magento/Sales/Model/AdminOrder/Create.php]: In this case we put value under key that will be processed on line 996. I have rewritten that code to be more readable.

File: app/code/Magento/Sales/Model/AdminOrder/Create.php (hunk 2 above)
Comment (SmVladyslav) [reply app/code/Magento/Sales/Model/AdminOrder/Create.php]: In this case we put value under key that will be processed on line 996. I have rewritten that code to be more readable.

**Comment by SmVladyslav** [comment]
@magento run all tests

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Integration Tests, WebAPI Tests

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B, Functional Tests  EE, Integration Tests

**Comment by SmVladyslav** [comment]
@magento run Functional Tests EE

File: app/code/Magento/Sales/view/adminhtml/web/order/create/scripts.js [hunk 3]
```diff
@@ -963,10 +964,18 @@ define([
             }.bind(this));
             // response handler
             productConfigure.setOnLoadIFrameCallback(listType, function (response) {
+                var area = ['items', 'shipping_method', 'billing_method', 'totals', 'giftmessage'];
```
Comment (zakdma) [inline app/code/Magento/Sales/view/adminhtml/web/order/create/scripts.js]: areas?

File: app/code/Magento/Sales/Test/Mftf/Test/CreateOrderFromEditCustomerPageTest.xml [hunk 4]
```diff
@@ -136,14 +136,15 @@
...
         <click selector="{{AdminCreateOrderWishListSection.addProductToOrderCheckBox($$simpleProduct.name$$)}}" stepKey="selectProductToAddToOrder"/>
+        <click selector="{{AdminCustomerCreateNewOrderSection.updateChangesBtn}}" stepKey="clickUpdateChangesButton"/>
         <click selector="{{AdminCreateOrderWishListSection.addConfigProductToOrder($$createConfigProduct.name$$)}}" stepKey="AddConfigurableProductToOrder"/>
         <waitForElementVisible selector="{{AdminOrderFormConfigureProductSection.optionSelect($$createConfigProductAttribute.default_frontend_label$$)}}" stepKey="waitForConfigurablePopover"/>
         <selectOption selector="{{AdminOrderFormConfigureProductSection.optionSelect($$createConfigProductAttribute.default_frontend_label$$)}}" userInput="$$getConfigAttributeOption1.label$$" stepKey="selectConfigurableOption"/>
         <click selector="{{AdminOrderFormConfigureProductSection.ok}}" stepKey="clickOkButton"/>
-        <click selector="{{AdminCustomerCreateNewOrderSection.updateChangesBtn}}" stepKey="clickOnUpdateButton"/>
+        <comment userInput="Click OK button to update lists" stepKey="clickOnUpdateButton"/>
```
Comment (zakdma) [inline app/code/Magento/Sales/Test/Mftf/Test/CreateOrderFromEditCustomerPageTest.xml]: userInput="Action should be removed but replaced with comment due to backward compatibility"

**Review by zakdma — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/view/adminhtml/web/order/create/scripts.js (hunk 3 above)
Comment (SmVladyslav) [reply app/code/Magento/Sales/view/adminhtml/web/order/create/scripts.js]: Renamed

File: app/code/Magento/Sales/Test/Mftf/Test/CreateOrderFromEditCustomerPageTest.xml (hunk 4 above)
Comment (SmVladyslav) [reply app/code/Magento/Sales/Test/Mftf/Test/CreateOrderFromEditCustomerPageTest.xml]: Fixed

**Comment by SmVladyslav** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31110`.

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario

1. Go to admin -> sales -> orders -> Created New Order -> Select Customer
2. Add simple product to the Order
3. Select Action => Move to Wish List
4. Click Update Items and Quantities button

5. Select checkbox 'Add To Order' => yes
6. Click Update Changes button


**Before applying fix:**
The product has been added to Cart but still remained in the _Wish List_
wish_list_before

**After applying fix**
Product moved to Order, wish List is empty

**Comment by lbajsarowicz** [comment]
@magento run Functional Tests B2B

**Pull request merged**